processor-api/src/main/java/com/synopsys/integration/alert/processor/api/filter/extractor/VulnerabilityNotificationExtractor.java

```diff
@@ -49,14 +48,27 @@ public VulnerabilityNotificationExtractor(Gson gson) {
     }
 
     @Override
-    public FilterableNotificationWrapper convertToFilterableNotificationWrapper(AlertNotificationModel alertNotificationModel, VulnerabilityNotificationContent vulnerabilityNotificationContent) {
-        List<String> affectedProjectNames = vulnerabilityNotificationContent.getAffectedProjectVersions()
-                                                .stream()
-                                                .map(AffectedProjectVersion::getProjectName)
-                                                .collect(Collectors.toList());
+    public List<FilterableNotificationWrapper> convertToFilterableNotificationWrapper(AlertNotificationModel alertNotificationModel, VulnerabilityNotificationContent vulnerabilityNotificationContent) {
         List<String> applicableSeverityTypes = new LinkedList<>(getApplicableSeverityTypes(vulnerabilityNotificationContent));
 
-        return FilterableNotificationWrapper.vulnerability(alertNotificationModel, vulnerabilityNotificationContent, affectedProjectNames, applicableSeverityTypes);
+        return breakDownIntoSeparateNotifications(vulnerabilityNotificationContent).stream()
+                   .map(vulnerabilitySingleProjectNotificationContent -> FilterableNotificationWrapper.vulnerability(
+                       // This leaves the AlertNotificationModel as the original but modifies the NotificationContent field
```

Comment:
Is this what we would want? Or would people expect the content returned here to exactly mimic the contents of AlertNotificationModel?

Reply:
We shouldn't need to use the `String content` from the `AlertNotificationModel` since the strongly-typed content is included.